### PR TITLE
doc/dev: update auto pr validation section

### PR DESCRIPTION
**Pull request by zdover23**
This commit updates the "Automated PR Validation"
section of the "Basic Workflow" page in the
Developer Guide. This is part of a project that
aims to clean all of the sentences currently extant
in the Dev Guide, prior to a slight reorganization
of that guide for the sake of clarity and simplicity.

Signed-off-by: Zac Dover <zac.dover@gmail.com>

## Checklist
- [ ] References tracker ticket
- [x] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by zdover23** [comment]
jenkins render docs

File: doc/dev/developer_guide/basic-workflow.rst [hunk 1]
```diff
@@ -221,21 +221,25 @@ as simple as::
...
-
-The :ref:`make-check`, builds the PR and runs it through a battery of
+automatically tests it. At the time of this writing (September 2020), the
+automated CI testing included five tests: 
+
+#. a test to check that the commits are properly signed (see :ref:`submitting-patches`): 
+#. a test to check that the documentation builds
+#. a test to check that the submodules are unmodified
+#. a test to check that the API is in order
+#. a make-check test
```
Comment (smithfarm) [inline doc/dev/developer_guide/basic-workflow.rst:233]: should we update this list?

Comment (smithfarm) [reply doc/dev/developer_guide/basic-workflow.rst:233]: "make-check" was a hyperlink before this change . . . could we preserve that?

File: doc/dev/developer_guide/basic-workflow.rst [hunk 2]
```diff
@@ -221,21 +221,25 @@ as simple as::
...
+
+#. a test to check that the commits are properly signed (see :ref:`submitting-patches`): 
+#. a test to check that the documentation builds
+#. a test to check that the submodules are unmodified
+#. a test to check that the API is in order
+#. a make-check test  
+   
+Additional tests may be performed depending on which files your PR modifies.
+
+The make-check builds the PR and runs it through a battery of
```
Comment (smithfarm) [inline doc/dev/developer_guide/basic-workflow.rst:237]: `The make-check test builds the PR ...`

and have `make-check` again be a hyperlink

Comment (zdover23) [reply doc/dev/developer_guide/basic-workflow.rst:237]: > i.e., insert the word "test" after "make-check" and have make-check again be a hyperlink

I've finally gotten back around to this PR. Check it out.

File: doc/dev/developer_guide/basic-workflow.rst (hunk 1 above)
Comment (zdover23) [reply doc/dev/developer_guide/basic-workflow.rst:233]: Sure thing, Nathan. I'll put it back, and I'll do so in a way that improves the doc.

Here's why I took it out in the first place: by default, the way that Sphinx creates hyperlinks involves inserting the text of the section that the link targets. This means that the sentence read "The What does "make check" mean test, builds", which is gibberish. Fortunately for all of us, I got the text to look the way I wanted and also preserved the hyperlink to the make-check section of the docs. So have another look and let me know if it looks right to you.

**Comment by zdover23** [comment]
@smithfarm: https://ceph--38450.org.readthedocs.build/en/38450/dev/developer_guide/basic-workflow/

I've restored the links and ensured that they don't break the structure of the sentences that they're in. If you get to this in the next day, great. If not, I'll use my new Durgin-granted powers to merge whatever I think should be merged. I'll do this in 12 to 24 hours.

**Review by jdurgin — APPROVED** [review]

**Review by smithfarm — APPROVED** [review]

**Pull request merged**